### PR TITLE
update player component

### PR DESCRIPTION
**Pull request by Drelarusoidd**

### CODE REVIEW

File: src/components/PlayerComponent.js [hunk 1]
```diff
@@ -1,79 +1,132 @@
...
-            document.getElementById(`play-btn-${num.index}`).classList.add("button");
+    const handleVolumeMute = () => {
+        const muted = audio.current.volume === 0
+        audio.current.volume = muted ? 1 : 0;
+        setMute(!muted)
+        volumePercent.current.style.height=audio.current.volume * 100 + '%';
+    }
+
+    const handleVolume = (event) => {
+        console.log('this');
```
Comment (ZubelRuslan) [inline src/components/PlayerComponent.js:51]: Don't forget to clean your code :)

File: src/components/PlayerComponent.js [hunk 2]
```diff
@@ -1,79 +1,132 @@
...
+        const response = await getPlaylist(track.playlist);
+        const [urlTrack, cover, trackName] = changeTrack('next', response, audio.current.src);
+        handleUpdatePlayer(urlTrack, cover, trackName);
+    }
+
+    const handlePreviousBtn = async() => {
+        const response = await getPlaylist(track.playlist);
+        const [urlTrack, cover, trackName] = changeTrack('previous', response, audio.current.src);
+        handleUpdatePlayer(urlTrack, cover, trackName);
+    }
```
Comment (ZubelRuslan) [inline src/components/PlayerComponent.js:75]:
```suggestion
    const handleChangeTrack = (rotation) = async() => {
        const response = await getPlaylist(track.playlist);
        const [urlTrack, cover, trackName] = changeTrack(rotation, response, audio.current.src);
        handleUpdatePlayer(urlTrack, cover, trackName);
    }
    const handleNextBtn = handleChangeTrack('next')
    const handlePreviousBtn = handleChangeTrack('previous')
```

File: src/components/PlayerComponent.js [hunk 3]
```diff
@@ -1,79 +1,132 @@
...
+
+        const progressbar = progress.current
+        progressbar.value = current_time / length;
+        if (player.currentTime === player.duration) {
+            setIsPlaying(false);
         }
     }
 
     return (
         <>
```
Comment (ZubelRuslan) [inline src/components/PlayerComponent.js:107]: Do you really need fragment here?

File: src/services/GetPlaylist.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import httpClient from "./httpClient";
+
+const getPlaylist = (playlistPk) => {
+    return httpClient.get(`playlist/${playlistPk}/`)
+        .then((response) => {
+            return JSON.stringify(response.data)
```
Comment (ZubelRuslan) [inline src/services/GetPlaylist.js:6]: Why do you need `stringify`?

File: src/utils/changeTrack.js [hunk 5]
```diff
@@ -0,0 +1,38 @@
+function changeTrack(method, data, currentUrl) {
+    let trackSet = JSON.parse(data)['track_set'];
```
Comment (ZubelRuslan) [inline src/utils/changeTrack.js:2]: `let` ?

File: src/utils/changeTrack.js [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    for (let i = 0; i < trackSet.length; i++) {
+      const url = trackSet[i]['track']
+      if (currentUrl === url) {
+        var currentIndex = i;
+        break;
+      }
+    }
```
Comment (ZubelRuslan) [inline src/utils/changeTrack.js:9]: `Array.findIndex`

File: src/utils/changeTrack.js [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+          const cover = trackSet[trackSet.length - 1]['cover'];
+          const trackName = trackSet[trackSet.length - 1]['name'];
+          return [url, cover, trackName]
+        }else {
+          const url = trackSet[currentIndex - 1]['track'];
+          const cover = trackSet[currentIndex - 1]['cover'];
+          const trackName = trackSet[currentIndex - 1]['name'];
+          return [url, cover, trackName]
+        }
+    }
```
Comment (ZubelRuslan) [inline src/utils/changeTrack.js:35]: Mb in the scope only set some target index variable and then after all if/else conditions extract `url, cover, trackName` only once?

**Review by ZubelRuslan — CHANGES_REQUESTED** [review]
This is not the end  ^_^

File: src/components/PlayerComponent.js [hunk 8]
```diff
@@ -1,79 +1,132 @@
...
-    
-        var totalLength = calculateTotalValue(length);
-        document.getElementsByClassName(`end-time-${num.index}`)[0].innerHTML = totalLength
-    
-        var currentTime = calculateCurrentValue(current_time);
-        document.getElementsByClassName(`start-time-${num.index}`)[0].innerHTML = currentTime
+    const handlePlayBtn = () => {
+        setIsPlaying(!isPlaying);
+        if (isPlaying) {
+            audio.current.pause();
```
[the remark this answers is not in the document]
Comment (Drelarusoidd) [reply src/components/PlayerComponent.js:37]: All HTMLAudioElement asynchronous.

[the remark this answers is not in the document]
Comment (ZubelRuslan) [reply src/components/PlayerComponent.js:37]: In this case.. just FYI -- will be cool to have some error handling for errs which can happen in this operations. There is possible to use promises functionality to adjust some additional logic and error handling

File: src/components/PlayerComponent.js (hunk 2 above)
Comment (Drelarusoidd) [reply src/components/PlayerComponent.js:75]: 'rotation' is not defined

Comment (ZubelRuslan) [reply src/components/PlayerComponent.js:75]: > 

Forgot to add `>`

**Review by ZubelRuslan — CHANGES_REQUESTED** [review]

**Review by ZubelRuslan — APPROVED** [review]
Congrads :)

**Pull request merged**